video_emotion/emotionTagger/face_emotion_extraction.py

```diff
@@ -1,61 +1,72 @@
 import os
-import numpy as np
+
+import cv2
+import numpy
 from keras.models import load_model
 import glob
 
-def get_labels(dataset_name):
-    if dataset_name == 'fer2013':
-        return {0:'angry', 1:'disgust', 2:'fear',
+def get_labels():
+    return {0:'angry', 1:'disgust', 2:'fear',
                 3:'happy', 4:'sad', 5:'surprise',
                 6:'neutral'}
-    else:
-        raise Exception('Invalid dataset name')
 
-def preprocess_input(x, v2=True):
+def preprocess_input(x):
     x = x.astype('float32')
     x = x / 255.0
-    if v2:
-        x = x - 0.5
-        x = x * 2.0
+    x = x - 0.5
+    x = x * 2.0
     return x
     
 def classify_face(face):
     # parameters for loading data and images
     dirname = os.path.abspath(os.path.dirname(__file__))
     emotion_model_path = os.path.join(
-        dirname,
-        "models/emotion_model.hdf5")
-    emotion_labels = get_labels('fer2013')
+                                    dirname,
+                                    "models/emotion_model.hdf5")
+    emotion_labels = get_labels()
 
     # loading models
     emotion_classifier = load_model(emotion_model_path)
 
     # getting input model shapes for inference
     emotion_target_size = emotion_classifier.input_shape[1:3]
 
-    face = cv2.cvtColor(face, cv2.COLOR_BGR2GRAY)
-    face = cv2.resize(face, (emotion_target_size))
-    face = preprocess_input(face, True)
-    face = np.expand_dims(face, 0)
-    face = np.expand_dims(face, -1)
+    face_grey = cv2.cvtColor(face, cv2.COLOR_BGR2GRAY)
+    face_resized = cv2.resize(face_grey, (emotion_target_size))
+    face_processed = preprocess_input(face_resized)
+    face_expanded = numpy.expand_dims(face_processed, 0)
+    face_expanded_more = numpy.expand_dims(face_expanded, -1)
     
-    emotion_prediction = emotion_classifier.predict(face) 
-    emotion_probability = np.max(emotion_prediction)
-    emotion_label_arg = np.argmax(emotion_prediction)
-    emotion_text = emotion_labels[emotion_label_arg]
+    emotion_prediction = emotion_classifier.predict(face_expanded_more)
     
-    #print("Mood: [" + str(emotion_labels[0]) + "]     Percentage: [" + str(emotion_prediction[0][0]) + "]")
-    #print("Mood: [" + str(emotion_labels[1]) + "]   Percentage: [" + str(emotion_prediction[0][1]) + "]")
-    #print("Mood: [" + str(emotion_labels[2]) + "]      Percentage: [" + str(emotion_prediction[0][2]) + "]")
-    #print("Mood: [" + str(emotion_labels[3]) + "]     Percentage: [" + str(emotion_prediction[0][3]) + "]")
-    #print("Mood: [" + str(emotion_labels[4]) + "]       Percentage: [" + str(emotion_prediction[0][4]) + "]")
-    #print("Mood: [" + str(emotion_labels[5]) + "]  Percentage: [" + str(emotion_prediction[0][5]) + "]")
-    #print("Mood: [" + str(emotion_labels[6]) + "]   Percentage: [" + str(emotion_prediction[0][6]) + "]")
-    return emotion_prediction[0]
+    return emotion_prediction[0] #emotion prediction gives an otherwise empty array
+def classify_faces(faces):
```

Comment:
Isn't this whole function using a lot of redundant code? Isn't it possible to use the ```def classify_face(face)```function multiple times instead? As in, redo the ```def classify_faces(faces)``` to use the singular version multiple times.

Reply:
No it's vastly slower